### PR TITLE
[WIP] Ingress docs

### PR DESCRIPTION
**Pull request by ijsnellf**

### CODE REVIEW

File: docs/tasks/ingress.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+... overview of ingress as a concept ...
+
+This task describes how to configure Istio to expose a service in a Kubernetes cluster. You'll learn how to create an Ingress controller, define a Ingress Resource and make requests to the service.
+
+In a Kubernetes environment, Istio uses [Kubernetes Ingress Resources](https://kubernetes.io/docs/concepts/services-networking/ingress/) to configure ingress behavior.   
+
+{% endcapture %}
+
+{% capture prerequisites %}
+* `kubectl` and access to a Kubernetes cluster with Istio deployed in it. See (xxx)[].
```
Comment (GregHanson) [inline docs/tasks/ingress.md]: possible copy paste error

**Comment by GregHanson** [comment]
don't forget to add ingress to the index task:
https://github.com/istio/istio.github.io/blob/master/_docs/tasks/index.md

File: docs/tasks/ingress.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+{% endcapture %}
+
+{% capture steps %}
+## Doing ...
+
+### Setup the environment
+Create an Istio Ingress controller.
+```yaml
+apiVersion: extensions/v1beta1
```
Comment (kyessenov) [inline docs/tasks/ingress.md]: @andraxylia is ingress installed by default? we can perhaps this step if it's a part of installation script.

File: docs/tasks/ingress.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+spec:
+  replicas: 1
+  template:
+    spec:
+      containers:
+      - name: app
+        image: <image name>
```
Comment (kyessenov) [inline docs/tasks/ingress.md]: use echo or httpbin service that Ed used.

File: docs/tasks/ingress.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  name: istio-ingress
+  annotations:
+    kubernetes.io/ingress.class: "istio"
+spec:
+  rules:
+  - http:
+      paths:
+      - path: /hello
+        backend:
+          serviceName: a
```
Comment (kyessenov) [inline docs/tasks/ingress.md]: serviceName should be helloworld.

File: docs/tasks/ingress.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+```bash
+$ curl http://192.168.99.100:80/hello
+.. response ..
+```
+{% endcapture %}
+
+{% capture discussion %}
+## Understanding ...
+
+Here's an interesting thing to know about the steps you just did.
```
Comment (kyessenov) [inline docs/tasks/ingress.md]: things that need to be covered here:
- using TLS in the ingress
- host-based routing

File: docs/tasks/ingress.md (hunk 2 above)
Comment (andraxylia) [inline docs/tasks/ingress.md]: Yes, istio-ingress is installed by default. Isaiah, you can add to prerequisites something like:

"Follow the Istio [Installation]({{site.bareurl}}/docs/tasks/istio-installation.html) guide".

BTW, I will rename istio-ingress-controller to istio-ingress.

Comment (ijsnellf) [inline docs/tasks/ingress.md]: Ok, thanks

**Comment by kyessenov** [comment]
Can we merge this to unblock iteration on the document?

**Comment by ijsnellf** [comment]
Sure. I don't have authorization to merge on this repo, though

**Pull request merged**